pkg/stream/enviroment.go

```diff
@@ -104,7 +104,11 @@ func (env *Environment) DeclareStream(streamName string, options *StreamOptions)
 	if err != nil {
 		return err
 	}
-	return client.DeclareStream(streamName, options)
+	errCreateStream := client.DeclareStream(streamName, options)
+	if errCreateStream == StreamAlreadyExists {
```

Comment:
So where does the precondition failed error come from? Does the server already return a response code for that and the client handles that into the precondition failed error?

I also wonder if the StreamAlreadyExists error shouldn't be handled from inside the client already. But if the server also returns that error itself, and the client is supposed to be a low-level interface for the protocol, I think it looks good like this.

Reply:
>So where does the precondition failed error come from?


from the server.

>I also wonder if the StreamAlreadyExists error shouldn't be handled from inside the client already.


I'd like to leave the TCPClient close to the protocol as much as possible. 
The `environment` is a smart client around the TCPClient

Reply:
Yeah agreed with keeping the client as low level as possible! Thanks for the clarifications